bn-apps/cordapp-updates-distribution/corda-updates-transport/src/main/kotlin/net/corda/cordaupdates/transport/flows/AbstractRepositoryHosterResponder.kt

```diff
@@ -0,0 +1,29 @@
+package net.corda.cordaupdates.transport.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.core.flows.FlowException
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+
+/**
+ * This class puts every incoming session through a [SessionFilter] if one has been specified in the configuration.
+ * In the case if no [SessionFilter] has been specified - every request will be let through.
+ *
+ * @throws FlowException if the incoming request doesn't satisfy [SessionFilter] requirements
+ *
+ * TODO: this logic should be implemented as flow overrides once the feature is available
+ */
+abstract class AbstractRepositoryHosterResponder<T>(val session : FlowSession) : FlowLogic<T>() {
+    @Suspendable
+    override fun call() : T {
+        val configuration = serviceHub.cordaService(RepositoryHosterConfigurationService::class.java)
+        val sessionFilter = configuration.getSessionFilter()
+        if (sessionFilter != null && !sessionFilter.isSessionAllowed(session, this)) {
+            throw FlowException("Counterparty ${session.counterparty} is not allowed to access repository")
+        }
+        return doCall()
+    }
+
+    @Suspendable
+    protected abstract fun doCall() : T
```

Comment:
I think you could just use `super.call()` in the sub-classes rather than add a new protected method here?

Reply:
Then I wouldn't be able to provide a return value to the `super.call()`. I.e.
```
abstract class MyAbstractClass<T> {
     abstract fun call(): T {
          doSomething()
          return // <- how would I return a result of a child's call() invocation?
     }
}
```

Reply:
Oh, I see. Fair enough then. Maybe we can find a better name than `doCall`? Maybe `postPermissionCheck`?

Reply:
Makes sense, renamed